addons/l10n_mn/migrations/1.1/pre-migrate.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+
+
+def migrate(cr, version):
+    ## Tax tags: Change the XMLids of tax tags which have been renamed.
+    tax_tag_changes = [
+        # Old tag            New tag
+        ('vat_report_tag38', 'vat_report_temp'),
+        ('vat_report_tag28', 'vat_report_tag27'),
+        ('vat_report_tag29', 'vat_report_tag28'),
+        ('vat_report_tag33', 'vat_report_tag32'),
+        ('vat_report_tag35', 'vat_report_tag34'),
+        ('vat_report_tag36', 'vat_report_tag35'),
+        ('vat_report_tag37', 'vat_report_tag36'),
+        ('vat_report_tag43', 'vat_report_tag38'),
+        ('vat_report_tag45', 'vat_report_tag40'),
+        ('vat_report_tag52', 'vat_report_tag54'),
+        ('vat_report_tag51', 'vat_report_tag53'),
+        ('vat_report_tag49', 'vat_report_tag51'),
+        ('vat_report_tag48', 'vat_report_tag50'),
+        ('vat_report_tag46', 'vat_report_tag48'),
+        ('vat_report_tag44', 'vat_report_tag46'),
+        ('vat_report_tag42', 'vat_report_tag45'),
+        ('vat_report_tag41', 'vat_report_tag44'),
+        ('vat_report_temp',  'vat_report_tag41'),
+    ]
+
+    queries = ['''
+        UPDATE ir_model_data
+        SET name = %s
+        WHERE name = %s
+            AND module = 'l10n_mn'
+            AND model = 'account.account.tag'
+    '''] * len(tax_tag_changes)
+    params = sum([[new_xmlid, old_xmlid] for old_xmlid, new_xmlid in tax_tag_changes], start=[])
+    cr.execute(';\n'.join(queries), params)
```

Comment:
You can do this in one query ;)
Untested example:
```suggestion
    tax_tag_changes = dict(tax_tag_changes)
    query = """
        UPDATE ir_model_data
           SET name = %s::jsonb->>name
         WHERE name IN %s
           AND module = 'l10n_mn'
           AND model = 'account.account.tag'
    """
    cr.execute(query, [json.dumps(tax_tag_changes), tuple(tax_tag_changes)]
```

Reply:
That's true. Actually I need two queries to avoid triggering the constraint that ensures that no two XMLids have the same name. But I really like the dict idea. Thanks.